src/plugins/ucc_discovery_plugin.cpp

```diff
@@ -243,12 +243,12 @@ void UccDiscoveryPlugin::_ucc_candidates_from_aggregate_node(std::shared_ptr<Abs
 void UccDiscoveryPlugin::_ucc_candidates_from_join_node(std::shared_ptr<AbstractLQPNode> node,
                                                         UccCandidates& ucc_candidates) {
   const auto& join_node = static_cast<JoinNode&>(*node);
-  // Fetch the join predicate to extract the UCC candidates from. Right now, limited to only equals predicates.
+  // Fetch the join predicate to extract the UCC candidates from. Right now, limited to single-predicate equi joins.
   const auto& join_predicates = join_node.join_predicates();
-  const auto binary_join_predicate = (join_predicates.size() == 1)
-                                         ? std::dynamic_pointer_cast<BinaryPredicateExpression>(join_predicates[0])
-                                         : std::shared_ptr<BinaryPredicateExpression>{};
-
+  if (join_predicates.size() != 1) {
+    return;
+  }
+  const auto binary_join_predicate = std::dynamic_pointer_cast<BinaryPredicateExpression>(join_predicates[0]);
```

Comment:
`join_predicates.front()`

Reply:
Was done by merging your suggestions in [251c0a8](https://github.com/hyrise/hyrise/pull/2485/commits/251c0a87e1c8bc67cc88b9e66d06c3b1a2d21d27)